httpcore5-h2/src/main/java/org/apache/hc/core5/http2/frame/DefaultFrameFactory.java

```diff
@@ -39,31 +39,35 @@
 public class DefaultFrameFactory extends FrameFactory {
 
     public static final FrameFactory INSTANCE = new DefaultFrameFactory();
+    /**
+     * A String for a {@code "Stream id"}.
+     */
+    private static final String STREAM_ID = "Stream id";
 
     @Override
     public RawFrame createHeaders(final int streamId, final ByteBuffer payload, final boolean endHeaders, final boolean endStream) {
-        Args.positive(streamId, "Stream id");
+        Args.positive(streamId, STREAM_ID);
```

Comment:
@arturobernalg Why? What is exactly the benefit of doing so?

Reply:
Hi @ok2c 

it's one of the sonar error-prone rules to make easy the refactor process.

https://rules.sonarsource.com/java/RSPEC-1192

Reply:
@arturobernalg I disagree. Sonar does a lot of questionable stuff and I do not see how this would simplify refactoring process at all.